File: videos/video.model.js
import mongoose from "mongoose";
const { Schema } = mongoose;

const videoSchema = new Schema({
  title: { type: String, required: true, maxLength: [100] },
  pageId: {
    type: Schema.Types.ObjectId,
    ref: "page",
    required: true,
  },
  videoUrl: { type: String, required: true },
  thumbnailUrl: { type: String, required: false },
  description: { type: String, required: false },
  createdAt: { type: Date, default: Date.now },
});

export default mongoose.model("video", videoSchema);

File: youtube_clone_app/lib/models/video_model.dart
class Video {
  final String id;
  final String title;
  final String videoUrl;
  final String thumbnailUrl;
  final String description;
  final int views;
  final DateTime createdAt;

  Video({
    required this.id,
    required this.title,
    required this.videoUrl,
    required this.thumbnailUrl,
    required this.description,
    this.views = 0,
    required this.createdAt,
  });

  factory Video.fromJson(Map<String, dynamic> json) {
    return Video(
      id: json['_id'],
      title: json['title'] ?? '',
      videoUrl: json['videoUrl'] ?? '',
      thumbnailUrl: json['thumbnailUrl'] ?? '',
      description: json['description'] ?? '',
      views: json['views'] ?? 0,
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt']) 
          : DateTime.now(),
    );
  }
}
